gguf-py/gguf/gguf_writer.py

```diff
@@ -282,9 +345,26 @@ def write_tensor_data(self, tensor: np.ndarray[Any, Any]) -> None:
 
         if self.endianess == GGUFEndian.BIG:
             tensor.byteswap(inplace=True)
-        self.write_padding(self.fout, self.fout.tell())
-        tensor.tofile(self.fout)
-        self.write_padding(self.fout, tensor.nbytes)
+
+        file_id = -1
+        for i, tensors in enumerate(self.tensors):
+            if len(tensors) > 0:
+                file_id = i
+                break
+
+        fout = self.fout[file_id]
+
+        # pop the first tensor info
+        # TODO: cleaner way to get the first key
+        first_tensor_name = [name for name, _ in zip(self.tensors[file_id].keys(), range(1))][0]
+        ti = self.tensors[file_id].pop(first_tensor_name)
+        assert len(ti.shape) == len(tensor.shape)
+        assert all(dim1 == dim2 for dim1, dim2 in zip(ti.shape, tensor.shape))
```

Comment:
These are too strict, unfortunately.

```suggestion
```

Otherwise `q8_0` conversion with `convert-legacy-llama.py` doesn't work, and same with modifying metadata on quantized models with `gguf-new-metadata.py`.

I think this has something to do with byte shapes vs quant shapes.

At least the `nbytes` check can be kept without problems.

Reply:
Oh, this is why we used to have a `get_tensor_size` function. Not all tensors passed from `convert-legacy-llama.py` are `np.ndarray[Any, Any]`.